src/realtime_metrics/main.py

```diff
@@ -175,73 +195,204 @@ def eta_accuracy(stop_time_updates: list[tuple[TripUpdate, StopTimeUpdate]]):
+    It computes the average amount of time in minutes a passenger has to wait at a stop, 
+    if he arrives at the arrival time of the most up-to-date stop time update.
+
+    Parameters:
+    stop_time_updates: list of corresponding trip updates and stop time updates
+
+    Returns:
+    A float containing the average wait time.
+    """
+    logger.info("Calculating Experienced Wait Time Delay...")
+
+    delays = []
+
+    logger.info(trip_stop_time_updates)
+
+    indexed_updates: Dict[tuple[str, str], list[tuple[TripUpdate, StopTimeUpdate]]] = {}
+
+    for key in actual_arrival_times.keys():
+        logger.debug("Key: %s", key)
+        route_id, _, stop_id = key
+        
+        new_key = (route_id, stop_id)
+
+        indexed_updates[new_key] = []
+
+    for trip_stop_time_update in trip_stop_time_updates:
+        trip_update: TripUpdate = trip_stop_time_update[0]
+        stop_time_update: StopTimeUpdate = trip_stop_time_update[1]
+
+        route_id = trip_update.route_id
+        stop_id = stop_time_update.stop_id
+        new_key = (route_id, stop_id)
+
+        if new_key not in indexed_updates.keys():
+            continue
+
+        indexed_updates[new_key].append((trip_update, stop_time_update))
+
+    logger.debug("Indexed updates: %s", indexed_updates)
+
+    for index_key, trip_stop_updates in indexed_updates.items():
+        if len(trip_stop_updates) < 2:
+            logger.debug("Not enough updates for %s", index_key)
+            continue
+        route_id, stop_id = index_key
+        # sort updates by published time
+        trip_stop_updates.sort(key=lambda u: u[0].timestamp.replace(tzinfo=timezone.utc).timestamp())
+        logger.debug("Sorted updates for %s: %s", index_key, [datetime.fromtimestamp(update[0].timestamp.replace(tzinfo=timezone.utc).timestamp()).strftime("%Y-%m-%d %H:%M:%S") for update in trip_stop_updates])
+
+        # Determine the range of days based on arrival times
+        min_arrival_time = trip_stop_updates[0][0].timestamp.replace(tzinfo=timezone.utc).timestamp()
+        max_arrival_time = trip_stop_updates[-1][0].timestamp.replace(tzinfo=timezone.utc).timestamp()
+
+        current_time = min_arrival_time - (min_arrival_time % 60)
+        end_time = max_arrival_time
+
+        for current_time in range(int(current_time), int(end_time) + 1, 60):
+            # find the last stop time update
+            next_predicted_arrival = get_last_predicted_update(current_time, trip_stop_updates)
+            if not next_predicted_arrival:
+                continue  # No prediction available
+            next_predicted_arrival_time = next_predicted_arrival[1].arrival_time
+            if next_predicted_arrival_time > current_time + 3600:
+                continue  # prediction is more than 60 minutes in the future
+            # find next experienced arrival at or after predicted 
+            next_actual_arrival = get_next_actual_arrival(next_predicted_arrival_time, route_id, stop_id)
```

Comment:
You claim that you expect ints, but you pass strings. As it seems to work, you might update the type hint in get_next_actual_arrival